source/Halibut/ServiceModel/PendingRequestQueue.cs

```diff
@@ -47,6 +47,12 @@ public ResponseMessage QueueAndWait(RequestMessage request, CancellationToken ca
             return pending.Response;
         }
 
+        public Task<ResponseMessage> QueueAndWaitAsync(RequestMessage request, CancellationToken cancellationToken)
```

Comment:
Can we make this one async so that the call stack is preserved if something goes 💥 ?